### PR TITLE
Pods should be managed by DaemonSets/Deployments

### PR DESCRIPTION
**Pull request by ydayagi**
Verify CNF pods' owners include Deployment or DaemonSet
https://issues.redhat.com/browse/CTONET-884

### CODE REVIEW

File: test-network-function/generic/suite.go [hunk 1]
```diff
@@ -208,6 +209,10 @@ var _ = ginkgo.Describe(testsKey, func() {
 		for _, containersUnderTest := range containersUnderTest {
 			testDeployments(containersUnderTest.oc.GetPodNamespace())
 		}
+
+		for _, containersUnderTest := range containersUnderTest {
+			testOwner(containersUnderTest.oc.GetPodNamespace(), containersUnderTest.oc.GetPodName())
```
Comment (crwr45) [inline test-network-function/generic/suite.go:214]: Out of interest, what's the reason for not using the existing `oc` context for the container?

File: test-network-function/generic/suite.go [hunk 2]
```diff
@@ -208,6 +209,10 @@ var _ = ginkgo.Describe(testsKey, func() {
 		for _, containersUnderTest := range containersUnderTest {
 			testDeployments(containersUnderTest.oc.GetPodNamespace())
 		}
+
+		for _, containersUnderTest := range containersUnderTest {
```
Comment (crwr45) [inline test-network-function/generic/suite.go:213]: I know there's several of these loops which are reusing `containersUnderTest` to refer to a single container inside the loop but please don't add another one. We'll clean up the others separately at a later time.

**Review by crwr45 — CHANGES_REQUESTED** [review]
One readability change, and one question. Otherwise this is great.

File: test-network-function/generic/suite.go (hunk 1 above)
Comment (ydayagi) [reply test-network-function/generic/suite.go:214]: I need admin access to the cluster and I do not need to execute anything within the pod.

File: test-network-function/generic/suite.go (hunk 2 above)
Comment (ydayagi) [reply test-network-function/generic/suite.go:213]: so merge into an existing loop?

Comment (ydayagi) [reply test-network-function/generic/suite.go:213]: I think I did not understand your first comment. What is the requested change?

Comment (crwr45) [reply test-network-function/generic/suite.go:213]: The term I was failing before to find was shadowing.
This `for` loop (and other similar ones in this file) is shadowing `containersUnderTest` inside the loop. This makes the code harder to understand and work with.
I'm not asking you to change all the occurrences of this (that should be done but will obfuscate the specifics of this change), rather not to add another one.

**Pull request merged**